// react-skinview3d/dist/index.js
import { useEffect, useRef } from "react";
import { SkinViewer } from "skinview3d";
import { jsx } from "react/jsx-runtime";

//#region src/index.tsx
/**
* React component that renders a 3D Minecraft skin viewer
* @param {ReactSkinview3dOptions} props - The component props
* @returns {React.ReactElement} A canvas element with the 3D skin viewer
*/
const ReactSkinview3d = ({ className, width, height, skinUrl, capeUrl, onReady, options }) => {
	const canvasRef = useRef(null);
	const skinviewRef = useRef(null);
	useEffect(() => {
		if (skinviewRef.current) return;
		const viewer = new SkinViewer({
			canvas: canvasRef.current,
			width: Number(width),
			height: Number(height),
			...options
		});
		skinUrl && viewer.loadSkin(skinUrl);
		capeUrl && viewer.loadCape(capeUrl);
		skinviewRef.current = viewer;
		if (onReady) onReady({
			viewer: skinviewRef.current,
			canvasRef: canvasRef.current
		});
	}, [
		width,
		height,
		skinUrl,
		capeUrl,
		onReady,
		options
	]);
	useEffect(() => {
		skinUrl ? skinviewRef.current.loadSkin(skinUrl) : skinviewRef.current.resetSkin();
	}, [skinUrl]);
	useEffect(() => {
		capeUrl ? skinviewRef.current.loadCape(capeUrl) : skinviewRef.current.resetCape();
	}, [capeUrl]);
	useEffect(() => {
		skinviewRef.current.setSize(Number(width), Number(height));
	}, [width, height]);
	return /* @__PURE__ */ jsx("canvas", {
		className,
		ref: canvasRef
	});
};

//#endregion
export { ReactSkinview3d };
