// react-skinview3d/dist/index.cjs
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
let react = require("react");
react = __toESM(react);
let skinview3d = require("skinview3d");
skinview3d = __toESM(skinview3d);
let react_jsx_runtime = require("react/jsx-runtime");
react_jsx_runtime = __toESM(react_jsx_runtime);

//#region src/index.tsx
/**
* React component that renders a 3D Minecraft skin viewer
* @param {ReactSkinview3dOptions} props - The component props
* @returns {React.ReactElement} A canvas element with the 3D skin viewer
*/
const ReactSkinview3d = ({ className, width, height, skinUrl, capeUrl, onReady, options }) => {
	const canvasRef = (0, react.useRef)(null);
	const skinviewRef = (0, react.useRef)(null);
	(0, react.useEffect)(() => {
		if (skinviewRef.current) return;
		const viewer = new skinview3d.SkinViewer({
			canvas: canvasRef.current,
			width: Number(width),
			height: Number(height),
			...options
		});
		skinUrl && viewer.loadSkin(skinUrl);
		capeUrl && viewer.loadCape(capeUrl);
		skinviewRef.current = viewer;
		if (onReady) onReady({
			viewer: skinviewRef.current,
			canvasRef: canvasRef.current
		});
	}, [
		width,
		height,
		skinUrl,
		capeUrl,
		onReady,
		options
	]);
	(0, react.useEffect)(() => {
		skinUrl ? skinviewRef.current.loadSkin(skinUrl) : skinviewRef.current.resetSkin();
	}, [skinUrl]);
	(0, react.useEffect)(() => {
		capeUrl ? skinviewRef.current.loadCape(capeUrl) : skinviewRef.current.resetCape();
	}, [capeUrl]);
	(0, react.useEffect)(() => {
		skinviewRef.current.setSize(Number(width), Number(height));
	}, [width, height]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)("canvas", {
		className,
		ref: canvasRef
	});
};

//#endregion
exports.ReactSkinview3d = ReactSkinview3d;
